nucleus/admin/config-api/src/main/java/org/glassfish/config/support/TranslatedConfigView.java

```diff
@@ -124,20 +117,43 @@ public static Object getTranslatedValue(Object value) {
             String origValue = stringValue;
             int i = 0;
             while (m.find() && i < MAX_SUBSTITUTION_DEPTH) {
-                String newValue = System.getProperty(m.group(2).trim());
+                String matchValue = m.group(2).trim();
+                String newValue = System.getProperty(matchValue);
                 if (newValue != null) {
                     stringValue = m.replaceFirst(
                             Matcher.quoteReplacement(m.group(1) + newValue + m.group(3)));
                     m.reset(stringValue);
-                }
+                } 
                 i++;     
             }
             if (i >= MAX_SUBSTITUTION_DEPTH) {
                 Logger.getAnonymousLogger().severe(
                         Strings.get("TranslatedConfigView.badprop", 
                         i, origValue));
             }
+            if (!stringValue.equals(origValue))
+                return stringValue;
```

Comment:
I'll test dropping down through to the env substitution code to see if that allows them to contain both otherwise it is an edge case for next release.